synapse/rest/__init__.py

```diff
@@ -113,6 +114,7 @@ def register_servlets(client_resource, hs):
         devices.register_servlets(hs, client_resource)
         thirdparty.register_servlets(hs, client_resource)
         sendtodevice.register_servlets(hs, client_resource)
+        shared_rooms.register_servlets(hs, client_resource)
```

Comment:
Hm, I may be inclined to combine the `filter` and `shared_rooms` servlets here as they both start with `/user`, but that could possibly be deferred to a later PR.

Reply:
Ah, good point. I'm minded to defer it to keep this change contained though.

Reply:
Sounds good to me.